.github/workflows/third-party-library-changes.yml

```diff
@@ -0,0 +1,78 @@
+name: Third Party Library Changes
+on:
+  pull_request:
+    branches:
+      - trunk
+      - "3.[89]"
+      - "[4-9].[0-9]"
+
+jobs:
+  third-party-changed-files:
+    runs-on: ubuntu-latest
+    name: Get third party library modified files
+    permissions:
+      pull-requests: read
+    outputs:
+      file_list: ${{ steps.modified_libraries.outputs.file_list }}
+      has_changed_files: ${{ steps.modified_libraries.outputs.has_changed_files }}
+    env:
+      THIRD_PARTY_LIBRARIES: (
+        "src/wp-includes/ID3"
+        "src/wp-includes/IXR"
+        "src/wp-includes/PHPMailer"
+        "src/wp-includes/Requests"
+        "src/wp-includes/SimplePie"
+        "src/wp-includes/sodium_compat"
+        "src/wp-includes/Text"
+        )
+    steps:
+      - name: Get changed files
+        id: changed_files
+        uses: tj-actions/changed-files@v37
+
+      - name: Get modified third party libraries
+        id: modified_libraries
+        run: |
+          ALL_CHANGED_FILES="${{ steps.changed_files.outputs.all_changed_files }}"
+          THIRD_PARTY_LIBRARIES=${{ env.THIRD_PARTY_LIBRARIES }}
+          THIRD_PARTY_CHANGED_FILES=()
+          HAS_CHANGED_FILES=false
+
+
+          for LIBRARY in "${THIRD_PARTY_LIBRARIES[@]}"; do
+            for FILE in $ALL_CHANGED_FILES; do
+              if [[ $FILE == $LIBRARY* ]]; then
+            	THIRD_PARTY_CHANGED_FILES+=("${FILE}")
+                HAS_CHANGED_FILES=true
+              fi
+            done
+          done
+
+          echo "file_list<<EOF" >> $GITHUB_OUTPUT
+          for LIBRARY in "${THIRD_PARTY_CHANGED_FILES[@]}"; do
+            echo "- $LIBRARY" >> $GITHUB_OUTPUT
+          done
+          echo 'EOF' >> $GITHUB_OUTPUT
+
+          echo "has_changed_files=${HAS_CHANGED_FILES}" >> $GITHUB_OUTPUT
+
+  comment-on-pull-request:
+    needs: third-party-changed-files
+    runs-on: ubuntu-latest
+    name: Comment on pull request with third party library changes
+    permissions:
+      pull-requests: write
+    if: ${{ needs.third-party-changed-files.outputs.has_changed_files == 'true' }}
+    steps:
```

Comment:
Instead of using a 3rd party action for this, could we instead utilize the actions/github-scripts action? There are a few other workflows in Core that use this that you can look to for examples.

Specifically, using the [create a review for a pull request](https://octokit.github.io/rest.js/v20#pulls-create-review) endpoint would be perfect for this workflow. The review should be submitted with the `REQUEST_CHANGES` event.